cli/azd/pkg/infra/provisioning/provider.go

```diff
@@ -21,10 +21,12 @@ type Options struct {
 	Provider ProviderKind `yaml:"provider"`
 	Path     string       `yaml:"path"`
 	Module   string       `yaml:"module"`
+	BicepADS bool
```

Comment:
I think we should add an explict yaml tag here so we know the casing that this filed is going to get in the yaml.  I think out of the box it is going to be `bicepads` based on something you previously taught me, @vhvb1989, which is probabbly not what we want?

Also - this value is specific to the bicep provider, I think this is our first instance of doing this - are we sure this is the right place to put this?

I guess another question is, do we need this right now?  In general, we don't expect folks would want to opt out of this at the project level right?  Do we even need support for disabling this in this way?

Reply:
My intention was not really to support this field from the `azure.yaml`.
I added here because this `options` structure is the one used to initialize the provider, and that's (sadly) the only moment where we can define the provider's state (because when provider is constructed, flags are not yet resolved by the cli, so we decided to create dummy providers and set their state by calling initialize).

Maybe there's a way to tell yaml, `ignore this field please, it's never expected to be found in the yaml`

Reply:
Consider adopting a more generic provider configuration approach here?  Instead of having something bicep specific on these root options we could introduce a generic `config` object of type `map[string]interface{}` that then is converted internally by the provider itself.  

This would be a similar concept to what I am doing for the remote environments in my PR @ https://github.com/Azure/azure-dev/pull/2699

Reply:
> Maybe there's a way to tell yaml, ignore this field please, it's never expected to be found in the yaml

There is.  You add the tag `yaml:"-'` (see https://pkg.go.dev/gopkg.in/yaml.v3#Marshal)